src/ql/pass/ana/visualize/detail/mapping.cc

```diff
@@ -275,16 +265,16 @@ void computeMappingPerCycle(const MappingGraphLayout &layout,
     }
 }
 
-Bool parseTopology(const ir::compat::PlatformRef &platform, Topology &topology) {
+Bool parseTopology(const ir::PlatformRef &platform, Topology &topology) {
     auto size = platform->topology->get_grid_size();
     if (size.x == 0 || size.y == 0) {
         QL_IOUT("Falling back on basic visualization; missing qubit coordinates in topology section");
         return false;
     }
     topology.xSize = size.x;
     topology.ySize = size.y;
-    topology.vertices.resize(platform->qubit_count, { 0, 0 });
-    for (utils::UInt q = 0; q < platform->qubit_count; q++) {
+    topology.vertices.resize(platform->qubits->shape[0], { 0, 0 });
```

Comment:
Need to discuss this. Does this work for any number of qubits? I'd expected a scan over the objects of qubits type and collecting them in some vector.

Reply:
As far as I understand, there was only one object of qubit type, referred to by platform->qubits, and this object should have the dimension of a vector (hence the shape[0]). See:
```
    # The main qubit register that the generic mapper will map everything to
    # and that topology applies to. The data type must be a vector of qubits.
    # This also indirectly defines the main qubit type.
    qubits: Link<physical_object>;
```

OR does the "main qubit register" formulation and your comment imply that there can be multiple?

I checked old_to_new.cc and new_to_old.cc and both seem to pair the number of qubits to platform->qubit_count (for old IR) and the 1st dimension of platform->qubits (for old IR) to the "qubit_number" json value in the platform config file. So in any case, in current code, `platform->qubit_count` should be equal to `platform->qubits->shape[0]`, I believe.